File: frontend/src/app/_models/user_favorite.ts
import { User } from './user';
import {Resource} from './resource';
import {Category} from './category';

export class UserFavorite {
  id: number;
  type: string;
  user_id: number;
  resource_id?: number;
  resource?: Resource;
  category_id?: number;
  category?: Category;
  age_range?: string;
  language?: string;
  covid19_category?: string;
  user?: User;

  constructor(private _props) {
    for (const propName in this._props) {
      if (this._props.hasOwnProperty(propName)) {
        this[propName] = this._props[propName];
      }
    }
  }
}

File: backend/app/model/user_favorite.py
from sqlalchemy import func, Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.orm import relationship

from app.database import Base
from app.model.category import Category
from app.model.resource import Resource
from app.model.user import User


class UserFavorite(Base):
    __tablename__ = "user_favorite"
    id = Column(Integer, primary_key=True)
    last_updated = Column(DateTime(timezone=True), default=func.now())
    type = Column(String)
    user_id = Column(Integer, ForeignKey(User.id), nullable=False)
    resource_id = Column(Integer, ForeignKey(Resource.id))
    category_id = Column(Integer, ForeignKey(Category.id))
    age_range = Column(String)
    language = Column(String)
    covid19_category = Column(String)
    user = relationship(User, backref="user_favorites")
    resource = relationship(Resource, backref="user_favorites")
    category = relationship(Category, backref="user_favorites")
